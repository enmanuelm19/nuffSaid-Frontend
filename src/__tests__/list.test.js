import React from "react";
import { mount } from "enzyme";
import { create } from "react-test-renderer";
import List from "../components/list";

describe("List", () => {
  const messages = [
    { message: "First message", priority: 1 },
    { message: "Second message", priority: 2 },
    { message: "Third message", priority: 3 },
  ];
  const component = mount(<List messages={messages} />);

  test("Render component", () => {
    expect(component.length).toEqual(1);
  });

  test("Render all messages", () => {
    const list = create(<List messages={messages} />);
    expect(list.toJSON()).toMatchSnapshot();
  });
});
