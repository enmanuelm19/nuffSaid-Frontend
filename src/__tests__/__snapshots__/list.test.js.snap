// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Render all messages 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-item MuiGrid-grid-xs-12"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
  >
    <h2
      className="MuiTypography-root MuiTypography-h5"
    />
    <h4
      className="MuiTypography-root MuiTypography-subtitle1"
    >
      Count 
      3
    </h4>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
  >
    <div
      className="MuiPaper-root MuiAlert-root MuiAlert-filledSuccess MuiPaper-elevation0"
      role="alert"
      style={
        Object {
          "backgroundColor": "#F56236",
        }
      }
    >
      <div
        className="MuiAlert-message"
      >
        First message
      </div>
      <div
        className="MuiAlert-action"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-1 MuiButton-colorInherit MuiButton-textSizeSmall MuiButton-sizeSmall"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Clear
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
  >
    <div
      className="MuiPaper-root MuiAlert-root MuiAlert-filledSuccess MuiPaper-elevation0"
      role="alert"
      style={
        Object {
          "backgroundColor": "#FCE788",
        }
      }
    >
      <div
        className="MuiAlert-message"
      >
        Second message
      </div>
      <div
        className="MuiAlert-action"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-1 MuiButton-colorInherit MuiButton-textSizeSmall MuiButton-sizeSmall"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Clear
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
  >
    <div
      className="MuiPaper-root MuiAlert-root MuiAlert-filledSuccess MuiPaper-elevation0"
      role="alert"
      style={
        Object {
          "backgroundColor": "#88FCA3",
        }
      }
    >
      <div
        className="MuiAlert-message"
      >
        Third message
      </div>
      <div
        className="MuiAlert-action"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-1 MuiButton-colorInherit MuiButton-textSizeSmall MuiButton-sizeSmall"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Clear
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
